generators/base/generator-base.mjs

```diff
@@ -225,66 +222,50 @@ export default class JHipsterBaseGenerator extends PrivateBase {
-    const packageJsonTargetFile = this.destinationPath('package.json');
-    const packageJsonTarget = this.fs.readJSON(packageJsonTargetFile);
-    const replace = section => {
-      if (packageJsonTarget[section]) {
-        Object.entries(packageJsonTarget[section]).forEach(([dependency, dependencyReference]) => {
-          if (dependencyReference.startsWith(keyToReplace)) {
-            const [keyToReplaceAtSource, sectionAtSource = section, dependencyAtSource = dependency] = dependencyReference.split('#');
-            if (keyToReplaceAtSource !== keyToReplace) return;
-            if (!packageJsonSource[sectionAtSource] || !packageJsonSource[sectionAtSource][dependencyAtSource]) {
-              throw new Error(`Error setting ${dependencyAtSource} version, not found at ${sectionAtSource}.${dependencyAtSource}`);
-            }
-            packageJsonTarget[section][dependency] = packageJsonSource[sectionAtSource][dependencyAtSource];
-          }
-        });
-      }
-    };
-    replace('dependencies');
-    replace('devDependencies');
-    this.fs.writeJSON(packageJsonTargetFile, packageJsonTarget);
+  isBuiltInEntity(entityName) {
+    return this.isBuiltInUser(entityName) || this.isBuiltInAuthority(entityName);
   }
 
   /**
    * @private
-   * Add a new icon to icon imports.
-   *
-   * @param {string} iconName - The name of the Font Awesome icon.
-   * @param {string} clientFramework - The name of the client framework
+   * Verify if the application is using built-in User.
+   * @return {boolean} true if the User is built-in.
    */
-  addIcon(iconName, clientFramework) {
-    if (clientFramework === ANGULAR) {
-      this.needleApi.clientAngular.addIcon(iconName);
-    } else if (clientFramework === REACT) {
-      // React
-      // TODO:
-    }
+  isUsingBuiltInUser() {
+    return isBuiltInUserConfiguration(this.jhipsterConfig);
   }
 
   /**
    * @private
-   * Add a new menu element, at the root of the menu.
-   *
-   * @param {string} routerName - The name of the Angular router that is added to the menu.
-   * @param {string} iconName - The name of the Font Awesome icon that will be displayed.
-   * @param {boolean} enableTranslation - If translations are enabled or not
-   * @param {string} clientFramework - The name of the client framework
-   * @param {string} translationKeyMenu - i18n key for entry in the menu
+   * Verify if the entity is a built-in User.
+   * @param {String} entityName - Entity name to verify.
+   * @return {boolean} true if the entity is User and is built-in.
    */
-  addElementToMenu(routerName, iconName, enableTranslation, clientFramework, translationKeyMenu = _.camelCase(routerName)) {
-    if (clientFramework === ANGULAR) {
-      this.needleApi.clientAngular.addElementToMenu(routerName, iconName, enableTranslation, translationKeyMenu, this.jhiPrefix);
-    } else if (clientFramework === REACT) {
-      // React
-      // TODO:
-    }
+  isBuiltInUser(entityName) {
```

Comment:
Those methods doesn't exists anymore at main

Reply:
removed, but please do not duplicate the tasks made here: rebasing is painful